packages/scratch-gui/src/components/library-item/library-item.jsx

```diff
@@ -36,10 +37,12 @@ class LibraryItemComponent extends React.PureComponent {
                             />
                         </div>
                     ) : null}
-                    <img
-                        className={styles.featuredImage}
-                        src={this.props.iconURL}
-                    />
+                    {this.props.iconSource ? (
+                        <ScratchImage
+                            className={styles.featuredImage}
+                            imageSource={this.props.iconSource}
+                        />
+                    ) : null}
```

Comment:
Future: I'd like to add general download-queue functionality somewhere central, maybe `scratch-storage`, and funnel these loads through that. Doing so should simultaneously obviate `ScratchImage` here _and_ make it so we can reliably load projects with thousands of assets without the browser freaking out. :muscle:

No action necessary right now, just sharing thoughts...